config.TEMPLATE.inc.php

```diff
@@ -282,6 +282,13 @@
 
 [email]
 
+; Enable email debugging, by default redirects email data to the email.log file under files folder
+; unless debug_path is specified
+; debug = On
+
+; Complete path to the file, where email data should be logged
+; debug_path = path
+
```

Comment:
I'd like to get @asmecher's thoughts on this, but looking at the config file we now have two sending options (log, sendmail), but the config file allows both of them to be configured at once.

Can we change the structure of our config file a bit to clarify that only option is available? Here's my proposal:

```
; Method to send email
; sendmail = Send emails using the sendmail binary on the server
; log = Write emails to a log in the files_dir
mailer = sendmail

; Optional. Override the default log path
; log_path =

; Path to sendmail
; The -bs argument is required for sending email with the SMTP protocol
sendmail_path = "/usr/sbin/sendmail -bs"

; Use SMTP
; smtp = On
```

Is that a little bit clearer? It also gives us an opportunity to expand support down the line. People could eventually set `mailer` to a third-party service liike Postmark, Mailgun, etc.

Reply:
Also, I couldn't find any uses of the following configuration options:

```
; Amount of time required between attempts to send non-editorial emails
; in seconds. This can be used to help prevent email relaying via OJS.
time_between_emails = 3600

; Maximum number of recipients that can be included in a single email
; (either as To:, Cc:, or Bcc: addresses) for a non-privileged user
max_recipients = 10
```

Should they be removed?

Reply:
Laravel allows having multiple sending options configured, including mentioned 3rd party services. Basically, each Mailable can configure its own way of being sent through them. I was thinking to preserve this, although don't have any practical use yet. The default setting is important here, as currently, all Mailables use the default transport. Maybe something like:
```
; Default method to send email
; sendmail = Send emails using the sendmail binary on the server
; log = Write emails to a log in the files_dir
; smtp
default = sendmail
```

Reply:
Laravel has an option `timeout` for SMTP transport, I'm not sure what it means. I suspect it's time before an email is sent, and not identical to `time_between_emails`, although can be used instead. It's only for SMTP though.

Reply:
Yeah, I like the `default` option. I think any other option would need to be coded into the configuration, though, so I think we'd probably want to prepare for them to have unique configurations too. I wonder if `mailer` would make more sense than `default` in this context. For example, we might want to separate out bulk emails into their own channel, and have those sent by a third-party API:

```
mailer = sendmail
mailer_bulk = postmark
```

Reply:
About logging: we don't have any kind of configurable log targets for the various kinds of logging in the system, yet, and I'd like to introduce that consistently and probably use the Laravel logging toolset when we do. So rather than allowing the specification of a log filename, could we just use the `error_log` for now? The system doesn't generate so many emails or other log messages (unless query logging is enabled) that this wouldn't be useful. (In my opinion!)

Reply:
I'm ok with that as an interim solution. However, I want to be able to use the mail logs to perform some integration tests (eg - is envelope sender / dmarc set? does this email only go to the assigned editor? etc), so having a dedicated log of some kind would be useful.

Reply:
> Could someone in theory write to any directory that the web server has access to?
Yes. But that was an idea - the plugin is for development purposes.